src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,84 @@
+import { useParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../utils/PersonLink';
+import classNames from 'classnames';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const { slugId } = useParams();
+  const findPersonByName = (name: string) => {
+    return people.find(person => person.name === name);
+  };
+
+  const peopleWithParents = people.map(person => {
+    const mother = person.motherName
+      ? findPersonByName(person.motherName)
+      : undefined;
+    const father = person.fatherName
+      ? findPersonByName(person.fatherName)
+      : undefined;
+
+    return { ...person, mother, father };
+  });
+
+  return (
+    <table
+      data-cy="peopleTable"
+      className="table is-striped is-hoverable is-narrow is-fullwidth"
+    >
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Sex</th>
+          <th>Born</th>
+          <th>Died</th>
+          <th>Mother</th>
+          <th>Father</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {peopleWithParents.map(person => {
+          return (
+            <tr
+              data-cy="person"
+              key={person.slug}
+              className={classNames({
+                'has-background-warning': slugId === person.slug,
+              })}
+            >
+              <td>
+                <PersonLink person={person} />
+              </td>
+
+              <td>{person.sex}</td>
+              <td>{person.born}</td>
+              <td>{person.died}</td>
+              <td>
+                {person.mother ? (
+                  <PersonLink person={person.mother} />
+                ) : person.motherName ? (
+                  person.motherName
+                ) : (
+                  '-'
+                )}
+              </td>
+              <td>
+                {person.father ? (
+                  <PersonLink person={person.father} />
+                ) : person.fatherName ? (
+                  person.fatherName
+                ) : (
+                  '-'
+                )}
```

Comment:
Code quality issue: The logic for rendering the father's name is quite complex and could be simplified. Consider creating a helper function to handle this, to make the code more readable.